src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -4148,23 +4148,148 @@ private Position endPosition()
       return Position.create(endRow, endColumn);
    }
    
+   // splits a chunk into two chunks
+   // chunk 1: first chunk line to linePos (not including linePos)
+   // chunk 2: linePos line to end
+   private void splitChunk(Scope chunk, int linePos)
+   {
+      Position chunkStart = chunk.getBodyStart();
+      Position chunkEnd = chunk.getEnd();
```

Comment:
You might need a guard here for chunks that have not yet been closed -- in such a case, I believe this could return `null`.